### PR TITLE
Deprecate merge_sets_safe in dojo/utils.py

### PR DESCRIPTION
**Pull request by manuel-sommer**
This PR uses python functionality which was added in newer python versions (3.6+) and thus deprecates method merge_sets_safe in utils.py. Furthermore, itertools is not needed anymore in utils.py. Thus, we don't need to import itertools.

### CODE REVIEW

**Comment by manuel-sommer** [comment]
I have no clue why this unittest fails.

**Comment by manuel-sommer** [comment]
Why is Kubernetes Deployment failing?

**Review by mtesauro — APPROVED** [review]
Approved

File: unittests/test_notifications.py [hunk 1]
```diff
@@ -53,4 +53,4 @@ def test_merge_notifications_list(self):
         self.assertEqual('alert' in merged_notifications.other, True)
         self.assertEqual('mail' in merged_notifications.other, True)
         self.assertEqual('slack' in merged_notifications.other, True)  # default alert from global
-        self.assertEqual(len(merged_notifications.other), 3)
+        self.assertEqual(len(merged_notifications.other), 4)
```
Comment (Maffooch) [inline unittests/test_notifications.py:56]: I would think functionality would not change if doing a simple todo. Why does this value increase?

**Review by Maffooch — CHANGES_REQUESTED** [review]

File: unittests/test_notifications.py (hunk 1 above)
Comment (manuel-sommer) [reply unittests/test_notifications.py:56]: This is also what I wrote in the comment above: _I have no clue why this unittest fails._

Comment (manuel-sommer) [reply unittests/test_notifications.py:56]: merged_notifications.other is now ['slack', 'mail', 'alert', 'alert']
before that it was: ['slack', 'mail', 'alert']

Comment (manuel-sommer) [reply unittests/test_notifications.py:56]: It is because of "default alert of global", see line 55 @Maffooch

**Comment by manuel-sommer** [comment]
Before the change: merged_notifications.other = {'alert', 'mail', 'slack'}
After the change: merged_notifications.other = ['slack', 'mail', 'alert', 'alert']

**Comment by kiblik** [comment]
> Before the change: merged_notifications.other = {'alert', 'mail', 'slack'}
> After the change: merged_notifications.other = ['slack', 'mail', 'alert', 'alert']
> I don't know exactly why alert is now duplicated.

In the first case, it is a set (no duplicates possible). In the second case, it is a list (duplicates allowed, order is important).
`[*..., *...]` concatnate 2 lists.

I suppose, we should use `{*..., *...}` in stread of `[*..., *...]`

```python
>>> a={'a','b'}
>>> b={'b','c'}
>>> c={*a, *b}
>>> c
{'a', 'b', 'c'}
>>> d=[*a, *b]
>>> d
['a', 'b', 'b', 'c']
```

**Comment by manuel-sommer** [comment]
@Maffooch, through the comment of @kiblik your requested change is resolved. 
Thank you @kiblik for the hint.

**Comment by mtesauro** [comment]
@manuel-sommer Got some feedback on this from the community/contributors:

Can you:
1. Add a description to the PR - why/what/etc
2. comment in the code to state what it is doing and why

Thanks!

**Comment by manuel-sommer** [comment]
@mtesauro, I added a description to the PR.

Furthermore, I have no clue why unittests now fails, it is not PR related, but looks like a false positive.

**Comment by manuel-sommer** [comment]
another closure to retrigger the pipeline, maybe this helps.

**Comment by manuel-sommer** [comment]
Reopen to retrigger pipeline

**Comment by manuel-sommer** [comment]
@Maffooch could you please review this PR?

**Review by Maffooch — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Pull request merged**